src/main/java/hotelserviceapp/sources/Booking.java

```
@@ -78,32 +87,37 @@ public int getNumberOfDays() {
 	 * Returns the booked room object.
 	 *
 	 * @return
+	 * Returns the bookedRoom.
 	 */
 	public Rooms getBookedRoom() {
 		return bookedRoom;
 	}
 
 	/**
-	 * Method for checking if the guest's EGN has a correct value.
+	 * Method for checking if the guest's ID has a correct value.
 	 *
-	 * @param EGN The guest's EGN.
+	 * @param ID The guest's ID.
 	 */
-	private void assertAndSetEGN(String EGN) {
-		if (EGN.length() < 10) {
-			System.out.println("ERROR :: Invalid EGN !");
-			guestEGN = "";
+
+	private void assertAndSetID(String ID) {
+		if (ID.length() < 10) {
+			System.out.println("ERROR :: Invalid ID !");
```

Let's throw a custom exception.